### DISCUSSION TITLE
利用Gradio主题生成器释放你的设计才华，赢得丰厚奖品！

### DISCUSSION THREAD

**Discussion by yvrjsharma** [GaiZhenbiao/ChuanhuChatGPT#531] [General]
🚨Attention All the ChuanhuChatGPT community members🚨
1️⃣ 🥳 In an exciting news, Gradio team has introduced a new tool called 🔥🔥Gradio-Theme-Builder🎨- This is a cool open-source Tool that makes creating stunning Themes for your Gradio apps super easy!
2️⃣ 🥳Gradio Themes Hackathon is extended until the end of this week - meaning there is a lot of time to explore and showcase your creativity and design skills to the community by building new themes!

💡 The Gradio Theme Builder is a Gradio app itself - Which is pretty cool in itself! 😍
This tool has made building themes super easy now. The tool is feature rich and gives you granular control on the process. You have endless possibilities for customization and creativity. The best part is that when you join the Hackathon Organisation and upload your unique Themes you stand a chance to win Amazing Prizes 🏆 and Huggingface Merchandise!

👉One way to access the Theme Builder at the Hackathon Org here - https://lnkd.in/dyUU7wuw
I used this hosted Theme-Builder tool to (1)create and (2)upload a "🦙Llamas theme" on 🤗Huggingface Spaces. Check out my Lllama🦙theme here - https://huggingface.co/spaces/ysharma/llamas - don't forget to toggle between the Light and Dark themes to truly appreciate the themes magic!!

👉Another way to access the Theme Builder is by downloading the latest version of Gradio & running the gr.theme_builder() command to access the tool locally.

🔥Use the amazing tool and create custom themes for any sector - finance, health, education, art, you name it!🌟
🚀We can't wait to see what unique & creative designs you will come up with!
Join Hackathon: https://huggingface.co/Gradio-Themes
Theme-Gallery: https://huggingface.co/spaces/gradio/theme-gallery
Theme-Guide: https://gradio.app/theming-guide/

**Comment by Keldos-Li** [comment]
Hi @yvrjsharma it's so nice of you to introduce gradio-theme-builder to us.

But we found the tool quite **limited**... For example, we cannot really change what messages look like in chatbot component, as well as other components...

Therefore we've made lots of effort to workaround. We created a css file and added variables for chatbot messages, and adjusted its padding, width, text-align to make the messages look neat and pretty. However, it looks like we _**are not able to**_ do the same thing in the HF theme "small and beautiful".

There is also a problem with sliders. Now the default style for sliders are so inconsistent in different browsers, and they look really ugly and outdated (especially in Safari). Pull request #734 shew the effort I made to build a prettier and consistent-looking slider.

Chuanhu Chat is a cool project with customized theme, it would be a pity if its theme is so bounded with the project itself in css and js files. Since we found you are a HF staff, maybe you can make it possible to make gradio theme builder more powerful with added parameters (I mean, parameters for chatbot messages especially)? And it would also be nice for gradio to adjust default style for sliders.
